### ISSUE TITLE
Feature request: Limit and from the end of the list

### ISSUE THREAD

**Issue by MarcelloTheArcane** [hasura/graphql-engine#3425]
If I understand correctly, you can use the `limit` parameter to specify only returning the first _`n`_ items. You can also use offset, which also fetches items from the start of the array.

However, I need to be able to fetch things from the end of the array, with an offset that works backwards (totally inverting the data).

See this handy diagram for an explanation: 

```
Data:
|  1 |  2 |  3 |  4 |  5 |  6 |  7 |  8 |  9 | 10 | 11 | 12 |

Limit: 5, Offset: 0
\____|____|____|____|____/

Limit: 5, Offset: 3
               \____|____|____|____|____/

Limit: 5, Offset: -3
|____|____/                                  \____|____|____|

Limit: -5, Offset 0
                                   \____|____|____|____|____/

Limit: -5, Offset: 3
|____|____|____/                                  \____|____|

Limit: -5, Offset: -3
                    \____|____|____|____|____/
```

I tried using negative numbers in the offset/limit, but apparently I can't. Would it be possible to add the functionality as described?

I suppose I could flip the `order_by`, and `.reverse()` the result as a workaround. However this will make my head hurt, and I'm sure would be sub-optimal for the webapp I am making.

**Comment by revskill10** [comment]
Why not sorting your data in reverse order then use `first` as normal ?

**Comment by MarcelloTheArcane** [comment]
@revskill10 

> However this will make my head hurt, and I'm sure would be sub-optimal for the webapp I am making.

If it's really hard to implement then no worries!

**Comment by igniteram** [comment]
@MarcelloTheArcane Did you find the optimum solution?

**Comment by MarcelloTheArcane** [comment]
@igniteram No, I didn't.  I can't actually remember now what I needed this for...
